app/frontend/src/service/api.ts

```diff
@@ -48,3 +51,23 @@ export async function ping() {
 
   return response.toObject();
 }
+
+export async function uploadFile(file: File): Promise<ImageInputValues> {
```

Comment:
Not sure if this manual work is necessary 🤔  We can always get the URL for this in a `useEffect` and populate the form's action that way?

I feel like it'd be simpler to use HTML form to handle this as it already deals with form data properly out of the box.

Reply:
I think it's better this way, since otherwise you would be making a media upload request every time you go to edit anything on your profile, plus an html form `submit()` doesn't return a promise, so it will end up more complicated.

Reply:
Does this need to be part of the main form though if this is a standalone thing? It shouldn't be sent every time if it's in its own form. What would you need the promise for in that case?